packages/um/package.py

```diff
@@ -331,3 +373,34 @@ def install(self, spec, prefix):
             install_bin_dir = join_path(prefix, bin_dir)
             mkdirp(install_bin_dir)
             install_tree(build_bin_dir, install_bin_dir)
+
+
+    def _dynamic_resource(self, url, ref, dst_dir):
+        """Check out resource dynamically based on a branch/tag/commit.
+
+        Parameters
+        ----------
+        url : str
+            Git URL.
+        ref : str
+            branch, commit or tag
+        dst_dir : str
+            Checkout path.
+        """
+
+        # Create the destination directory
+        mkdirp(dst_dir)
+
+        git = spack.util.git.git()
+
+        # Attempt to check out branch to dir
+        try:
+            tty.msg(f"Attempting to checkout branch {ref}")
+            git("clone", "--depth", "1", "--branch", ref, url, dst_dir)
+        except ProcessError:
+            tty.warn(f"ref '{ref}' may be a commit/tag, retrying.")
+            git("clone", url, dst_dir)
+            with working_dir(dst_dir):
+                git("checkout", ref)
```

Comment:
Apparently it should be fine to `git clone` a commit or tag. Have you seen `git` throw a `ProcessError` in this case?